### DISCUSSION TITLE
add @ to profile display of handle

### DISCUSSION THREAD

**Discussion by wakest** [cheeaun/phanpy#483] [Ideas]
in the profile it shows usernames at name@domain.tld but then when you click copy it adds an @ before it. and you need an @ to use it anywhere so it seems weird not to show it.
<img width="511" alt="image" src="https://github.com/cheeaun/phanpy/assets/7890201/4436c07a-ffda-41fd-b036-0e0032603330">

in the ... menu it lists the handle in the double @@ form (in what I would consider the correct way)
<img width="325" alt="image" src="https://github.com/cheeaun/phanpy/assets/7890201/1f03c457-4894-4cd8-8e89-da6f8cb35df9">

I know this is a touchy subject that many people have strong feelings about, so mostly just opening this issue to discuss the reasons behind the choice either way. I would say it should be normalized across Phanpy regardless of the decision

**Comment by cheeaun** [comment]
Converted this to a discussion.

### Relevant links

- https://mastodon.social/@cheeaun/112099567913409079 
  - Related input on initial '@' from Blaine (founding engineer of Twitter) https://mastodon.social/@blaine/112100691575688060
- https://blog.joinmastodon.org/2024/02/mastodon-for-android-update/

### My thoughts/research

- Initial '@' is kinda syntactical or a visual hint
  - Has become a norm thanks to Twitter and widely used in every single social media (and chat) platforms.
  - Early designs of Twitter did not show '@' except for mentions https://twirpz.wordpress.com/2015/06/03/the-history-of-twitter-in-profile-pages-2006-to-2015/
    - It's functional, to at-mention someone, not part of a username
    - It uses more '@' when the newer designs start showing account display names beside the username (`Your name @username`), which I suspect the designers were using '@' to visually differentiate names from usernames.
  - The only current social networks that I'm aware of that don't show '@' is Instagram and Threads.
    - This is probably due to their UI **not** showing display names in the timeline. Instead of `Your name @username`, they show just `username`.
    - This is interesting because no one's complaining and people still know how to at-mention.
- As of now (March-April 2024), Phanpy experimentally removed initial '@' from remote accounts (`username@domain` instead of `@username@domain`). Local accounts will still have initial '@' (`@username` - without domain)
  - Mastodon has this "convention" that if it's local instance, show `@cheeaun`. If remote, show `@cheeaun@otherdomain.social`. This also applies to at-mentions when composing posts, which I think is quite confusing especially for beginners. If you're on `mastodon.social`, just typing `@cheeaun` will automagically refer to `@cheeaun@mastodon.social` instead of `@cheeaun@otherdomain.social`.
  - Mammoth (iOS) app does this too.

### Other relevant(?) discussions

- Lemmy uses exclamation mark (`!community@domain`) https://lemmy.eus/comment/116730
  - and possibly other characters? 🫠 https://github.com/LemmyNet/lemmy/issues/2987
- Threads implement "hashtags" without hash https://techcrunch.com/2023/11/15/threads-starts-testing-hashtags-without-the-hash/

**Comment by shadowwwind** [comment]
Removing the @ makes it confusing as it could be an email address as well. I like the @.
